### DISCUSSION TITLE
New installs broken on Pi Zero W

### DISCUSSION THREAD

**Discussion by Hajman** [mKeRix/room-assistant#957] [Q&A]
I installed on a new Pi Zero about 2 weeks ago.  Nodejs LTS at that point was 14.8.1.  The software ran fine though with inaccurate presence detection.  So I got another zero and tried to install this weekend.  Everything installs but room-assistant throws an error.  It isn't nodejs version related as I dropped back to 14.8.1 (known to work properly) and the same error occurs.  I think the dependency of command-line-usage changed given the below message.

/opt/nodejs/lib/node_modules/room-assistant/node_modules/command-line-usage/index.js:1
rix_keypad_remove

SyntaxError: Invalid or unexpected token
    at wrapSafe (internal/modules/cjs/loader.js:1001:16)
    at Module._compile (internal/modules/cjs/loader.js:1049:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:93:18)
    at Object.<anonymous> (/opt/nodejs/lib/node_modules/room-assistant/bin/room-assistant.js:5:26)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)

**Comment by Arkadije** [comment]
When I try to install it I get the following error:

```
pi@raspberrypi:~ $ sudo npm i --global --unsafe-perm room-assistant
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated uuid@3.4.0: Please upgrade to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated tar@2.2.2: This version of tar is no longer supported, and will not receive security updates. Please upgrade asap.

added 327 packages, and audited 328 packages in 36m

21 packages are looking for funding
  run `npm fund` for details

16 vulnerabilities (12 moderate, 4 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
```

Hope someone can resolve this issue.

**Comment by MrTechGadget** [comment]
@Hajman I have built Pi Zero W's in the past two days, both working fine. If a sudo apt update && sudo apt upgrade doesn't fix it for you, I would suggest rebuilding from scratch.

@Arkadije those are not errors, those are warnings. room-assistant still runs fine.

**Reply by Hajman** [reply]
Hmm.  I started with a freshly imaged SSD card, updated the PiOS (sudo apt update && sudo apt full-upgrade) and follow all of the steps twice.  It was the same procedure that I followed for the original Pi Zero W that worked fine.  Ultimately, I just cloned the SSD with the working instance to 4 additional instances.  Now I am working to get more accurate room to room detection.

**Comment by Arkadije** [comment]
Ths for replying back. I solved the issue. My problem was when I run the room-assistant command to start it, it gave me the error, no such file.
That is why I thought that something was wrong. 
But I checked the files that are used for generating the service, and the rest is history :D
